src/docfx/restore/RestoreMap.cs

```diff
@@ -50,8 +54,9 @@ string FindLastModifiedGitRepository()
                 return (
                     from path in Directory.GetDirectories(repoPath, "*", SearchOption.TopDirectoryOnly)
                     let name = Path.GetFileName(path)
-                    where name.StartsWith(HrefUtility.EscapeUrlSegment(branch) + "-" + branch.GetMd5HashShort() + "-") &&
-                          GitUtility.IsWorkTreeCheckoutComplete(repoPath, name)
+                    where GitUtility.IsWorkTreeCheckoutComplete(repoPath, name) &&
+                        ((locked && name == $"{RestoreGit.GetWorkTreeHeadPrefix(branch, locked)}-{commit}") ||
+                        (!locked && name.StartsWith(RestoreGit.GetWorkTreeHeadPrefix(branch))))
                     orderby new DirectoryInfo(path).LastWriteTimeUtc
```

Comment:
This orderby may not be correct, we may need to ask git for the latest version, it can be done when doing the slot/rent/return model.

Reply:
I've spited them into {branch}-xxxx and locked-{branch}-xxx models, make sure the original behavior is correct(last write time order)